tests/test_message_provider.py

```diff
@@ -0,0 +1,162 @@
+import os
+# import pytest
+from mock import patch, Mock
+from unittest import TestCase
+
+from pact.message_provider import MessageProvider
+from pact import message_provider as message_provider
+
+class MessageProviderTestCase(TestCase):
+    def _mock_response(
+            self,
+            status=200,
+            content="fake response",
+            raise_for_status=None):
+
+        mock_resp = Mock()
+        mock_resp.raise_for_status = Mock()
+        if raise_for_status:
+            mock_resp.raise_for_status.side_effect = raise_for_status
+
+        mock_resp.status_code = status
+        mock_resp.text = content
+        return mock_resp
+
+    def message_handler(self):
+        return {'success': True}
+
+    def setUp(self):
+        self.provider = MessageProvider(
+            provider='DocumentService',
+            consumer='DetectContentLambda',
+            message_providers={
+                'a document created successfully': self.message_handler
+            }
+        )
+
+    def test_init(self):
+        self.assertIsInstance(self.provider, MessageProvider)
+        self.assertEqual(self.provider.provider, 'DocumentService')
+        self.assertEqual(self.provider.consumer, 'DetectContentLambda')
+        self.assertEqual(self.provider.pact_dir, os.getcwd())
+        self.assertEqual(self.provider.version, '3.0.0')
+        self.assertEqual(self.provider.proxy_host, 'localhost')
+        self.assertEqual(self.provider.proxy_port, '1234')
+
+    @patch('pact.Verifier.verify_pacts', return_value=(0, 'logs'))
+    def test_verify(self, mock_verify_pacts):
+        self.provider.verify()
+
+        assert mock_verify_pacts.call_count == 1
+        mock_verify_pacts.assert_called_with(f'{self.provider.pact_dir}/{self.provider._pact_file()}', verbose=False)
+
+class MessageProviderContextManagerTestCase(MessageProviderTestCase):
+    def setUp(self):
+        super(MessageProviderContextManagerTestCase, self).setUp()
+
+    @patch('pact.MessageProvider._start_proxy', return_value=0)
+    @patch('pact.MessageProvider._stop_proxy', return_value=0)
+    def test_successful(self, mock_stop_proxy, mock_start_proxy):
+        with self.provider:
+            pass
+
+        mock_start_proxy.assert_called_once()
+        mock_stop_proxy.assert_called_once()
+
+    @patch('pact.MessageProvider._wait_for_server_start', side_effect=RuntimeError('boom!'))
+    @patch('pact.MessageProvider._start_proxy', return_value=0)
+    @patch('pact.MessageProvider._stop_proxy', return_value=0)
+    def test_stop_proxy_on_runtime_error(self, mock_stop_proxy, mock_start_proxy, mock_wait_for_server_start,):
+        with self.provider:
+            pass
+
+        mock_start_proxy.assert_called_once()
+        mock_stop_proxy.assert_called_once()
+
+    # @pytest.mark.skip(reason="Caused BrokenPipeError in PactStartShutdownServerTestCase::test_stop_fails_posix. Need to investigate")
```

Comment:
I think this caused by Flash. Now that it's replaced with FastAPI, I think we can remove the commented code

Reply:
Done